### ISSUE TITLE
How to remove iOS permissions after cocoapods is installed?

### ISSUE THREAD

**Issue by rolinger** [lemberg/ios-permissions-service#17]
NOTE:  i realize I am writing this in the ios-permissions-service` - half of this issue should be for here, the other half should be for the cocoapods team.

I am using `cordova-plugin-firebasex` with my cordova app - per its instructions I have installed cocoapods and its dependencies.  I guess the default pods install includes access to a bunch of iOS permissions that my app doesn't need (7 of them) and now my app is getting rejected because of missing or insufficient permissions/usage-descriptions for these 7 permissions.

> ITMS-90683: Missing Purpose String in Info.plist - Your app's code references one or more APIs that access sensitive user data. The app's Info.plist file should contain a NSContactsUsageDescription key with a user-facing purpose string explaining clearly and completely why your app needs the data. Starting Spring 2019, all apps submitted to the App Store that access user data are required to include a purpose string. If you're using external libraries or SDKs, they may reference APIs that require a purpose string. While your app might not use these APIs, a purpose string is still required. You can contact the developer of the library or SDK and request they release a version of their code that doesn't contain the APIs. Learn more

 I see all the permissions in the `Pods-myApp-Info.plist` file and then later read that I could use the `PermissionsConfiguration.xcconfig` file in the root of my project.  So I created the file, commented out all the permissions I don't want added, copied it my project root and also placed a copy in `../myApp/platforms/ios` directory - I then ran `pod update` but it didn't work.

How can I remove these from my project?  I need to get my next version of my app submitted asap and I don't want to leave these permissions in my project for a variety of obvious reasons. 
 Can you please provide documentation, or create it, to remove these permissions once they get installed?   

For future, maybe the install process should ask the user which permissions to turn on - simply adding them all when the majority aren't needed.  The user finding out AFTER the install that they should have added the `PermissionsConfiguration.xcconfig` to prevent these permissions from being added is a bit backward...and in my case, a bit too late.

**Comment by OverSwift** [comment]
Hi @rolinger 
as I know when you using cocoa pods you can specify concentrate permission to use and only this sources will be downloaded and added to your project
On screen shoot below I only added a `Location` permissions to my test project and only this sources was added.
<img width="958" alt="Знімок екрана  о 18 29 33" src="https://user-images.githubusercontent.com/1131206/63604246-f58de800-c5d3-11e9-93ef-20cb036cee62.png">

**Comment by rolinger** [comment]
@OverSwift   Yeah, thats whats messed up, here is my pod file:
```
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '9.3'
use_frameworks!
target 'myApp' do
	project 'myApp.xcodeproj'
	pod 'Firebase/Core', '6.3.0'
	pod 'Firebase/Auth', '6.3.0'
	pod 'Firebase/Messaging', '6.3.0'
	pod 'Firebase/Performance', '6.3.0'
	pod 'Firebase/RemoteConfig', '6.3.0'
	pod 'Fabric', '1.9.0'
	pod 'Crashlytics', '3.12.0'
end
```

but yet in my `Pods-myApp-Info.plist` file in the `myApp.xcworkspace` this is added:
```
	<key>NSBluetoothPeripheralUsageDescription</key>
	<string>This app requires bluetooth access to function properly.</string>
	<key>NSCameraUsageDescription</key>
	<string>This app requires camera access to function properly.</string>
	<key>NSPhotoLibraryUsageDescription</key>
	<string>This app requires photo library access to function properly.</string>
	<key>NSMicrophoneUsageDescription</key>
	<string>This app requires microphone access to function properly.</string>
	<key>NSContactsUsageDescription</key>
	<string>This app requires contacts access to function properly.</string>
	<key>NSCalendarsUsageDescription</key>
	<string>This app requires calendar access to function properly.</string>
	<key>NSRemindersUsageDescription</key>
	<string>This app requires reminders access to function properly.</string>
	<key>NSMotionUsageDescription</key>
	<string>This app requires motion detection access to function properly.</string>
```

And those are the 7 permissions my app is being rejected for.

**Comment by rolinger** [comment]
@OverSwift - so any method to remove all these pod added iOS requirements without starting my pod/project installation over?

**Comment by OverSwift** [comment]
Hi @rolinger, made some more few tests and as I saw library do not modify Pods-myApp-Info.plist at all. Also try with list of your libraries. Maybe try `pod deintegrate && pod install && pod update` to reinstall the pods

**Comment by rolinger** [comment]
@OverSwift - I found the source of all the iOS permissions being added.  Its from a plugin (cordova.plugins.diagnostics) I have used for years that made significant changes and is now auto-adding those permissions unless you specify before not to include them before the install.  It never occurred to me this is where they were coming from because the only change I made in my current version of the app was adding `cordova-plugin-firebasex` which relies heavily on Cocoapods - and then seeing all the permissions added to the `Pods-myApp-Info.plist` file I just assumed the additional permissions were coming from Cocoapods.  Sorry about all that.   What I didn't realize is my current version of my app I did a rebuild on it and the rebuild auto-included the latest version of the `diagnostics` plugin which brought all these changes.  The previous version of that plugin did not do this.
